crypto/s2n_ecc_evp.c

```diff
@@ -469,6 +463,18 @@ int s2n_ecc_evp_write_params(struct s2n_ecc_evp_params *ecc_evp_params, struct s
     /* key share + key share size (1) + iana (2) + curve type (1) */
     written->size = key_share_size + 4;
 
+    /* Write data after writing ecc evp params to avoid stuffer resize failure */
+    uint32_t read_cursor = out->read_cursor;
+    uint32_t written_data_location = s2n_stuffer_data_available(out) - written->size;
+
+    POSIX_GUARD(s2n_stuffer_rewrite(out));
+    POSIX_GUARD(s2n_stuffer_skip_write(out, written_data_location));
+    POSIX_GUARD(s2n_stuffer_skip_read(out, read_cursor));
```

Comment:
This has the potential not to work. You can't read past where you've written, so this only works if read_cursor < written_data_location. I imagine it works bc read_cursor should be 0 here, but it's not technically correct.

It seems non-ideal to have to deal with the read_cursor at all. Can you figure out a less destructive way to handle this?